docs/builds.md

```diff
@@ -1,76 +1,31 @@
-jobs:
-  # This workflow contains a single job called "build"
-  build:
-    # The type of runner that the job will run on
-    runs-on: ubuntu-latest
+Let's at a common feature branching example:
 
-    # Steps represent a sequence of tasks that will be executed as part of the job
-    steps:
-      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
-      - uses: actions/checkout@v2
+![Builds are associated by putting Jira issue keys into commit messages](./images/associating-builds.png)
 
-      # Runs a single command using the runners shell
-      - name: Run a one-line script
-        run: echo Hello, world!
+We have two feature branches off the `main` branch. Say we have configured our GitHub Actions so that each commit on each of the feature branches (green dots) and each commit on the `main` branch (blue dots) triggers a workflow to build and test our codebase.
 
-      # Runs a set of commands using the runners shell
-      - name: Run a multi-line script
-        run: |
-          echo Add other actions to build,
-          echo test, and deploy your project.
-          sleep 60s
-```
+If we use the Jira issue keys (`JIRA-*`) in the commit messages as shown in the diagram, we would see workflow run #6 in the Jira issues `JIRA-1` and `JIRA-2` and workflow run #7 in the Jira issues `JIRA-1`, `JIRA-10`, and `JIRA-11`.
 
-Once you have a similar file in any of your repositories that are connected to your Jira via the app, you will start to see builds data
-in the development panel in Jira.
+Workflow run #2 would be associated with the Jira issue `JIRA-1` only. Workflow run #4 would be associated with both `JIRA-1` and `JIRA-2`, because they're both part of the same pull request.
 
-![Builds data in Jira](./images/builds-data-jira-dev-panel.png)
+**IMPORTANT:** The app only looks for Jira issue keys in commits that were part of a pull request. In the above example, all commits on the `feature-1` and `feature-2` branches are part of a respective pull request into the `main` branch. 
 
-#### Supporting Multiple Commits with Issue Keys
+**If you merge a branch into another branch without a pull request, only the last commit message on that branch will be searched for Jira issue keys**. 
 
-One important thing to note in the above example is the branches being targeted under `pull_requests` `branches`:
+To run your GitHub workflows on pull requests, make sure to include the branch name pattern of your pull request source branches in your GitHub workflow config like this:
 
-```
-pull_request:
+```yaml
+on:
+  pull_request:
     branches:
       - main
-      - feature/**
+      - feature**
 ```
-
-When you open a pull request in a connected repository, the GitHub for Jira app can compare two points in your git history is we have access to a base branch and a head branch. If the app has both of these, it can make a request to GitHub to compare your
```

Comment:
I made the explanation a lot less detailed because the details might confuse people (they confused me, at least :))